### PR TITLE
Completed test suite

### PR DESCRIPTION
**Pull request by LisaCarlson**

### CODE REVIEW

File: spec/grader_spec.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (craftninja) [inline spec/grader_spec.js]: Watch your whitespace... errors in multiple lines (40, 59, 62, 65, 68, 71) and only need one empty line below code (75 is fine, but 76 - 93 should be deleted). Also, empty lines between `it` blocks (like you are doing with `describe` blocks) would be totally fine if that helps readability.

File: spec/grader_spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  }); 
+  it('returns the correct median score even if the array elements are unordered', function(){
+    expect(code.medianScore([99, 82, 92, 95, 100, 54, 76])).toEqual(92);
+  }); 
+
+});
+
+describe('Mode Score', function() {
+  it('takes an array of test scores and calculates the mode score', function(){
+  expect(code.modeScore([82, 92, 75, 91, 92, 89, 95, 100, 86])).toEqual(92);
```
Comment (craftninja) [inline spec/grader_spec.js]: Feel free to refactor variables out of your tests when they take up too much space. Just make sure they have good names!

File: grader.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  else if (input >= 70 && input <= 79) {
+    return 'C';
+  }
+  else if (input >= 60 && input <= 69) {
+    return 'D';
+  }
+  else if (input >= 0 && input <= 59) {
+    return 'F';
+  }
+},
```
Comment (craftninja) [inline grader.js]: How could you refactor this to not need the && statements?

File: grader.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+averageScore: function(array) {
+  if (array === undefined) {
+    return '[]';
+  }
+  var sum = array.reduce(function(previous, current){
+   return previous + current; 
+  });
+  return isNaN(sum) ? '[]' : sum / array.length;    
+},
```
Comment (craftninja) [inline grader.js]: I do like what you are thinking here, but is out of scope for the requirements. Definitely ask your project manager what do to in these edge cases!

File: grader.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+  function checkCount(array, value) {
+    var check = 0;
+    for (var j = 0; j < array.length; j++) {
+      if (value == array[j]) {
+        check += 1;
+      }
+    }
+  return check;
```
Comment (craftninja) [inline grader.js]: whitespace

File: grader.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+medianScore: function(array) {
+  if (array === undefined) {
+    return '[]';
+  }
+  array = array.sort(function(a, b){
+    return (a > b) ? 1 : ((a < b) ? -1 : 0);
+  });
+  var isNumber = array.reduce(function(previous, current){
+   return previous + current;
```
Comment (craftninja) [inline grader.js]: whitespace

File: grader.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+
+}
+
+
+
+
+
+}
```
Comment (craftninja) [inline grader.js]: No need for all these empty lines.

**Comment by craftninja** [comment]
Watch for adding solutions to edge cases not mentioned in the exercises (although I do like a lot of the "questions" you addressed!). Also think about making your test descriptions more succinct. Careful on your whitespacing!

**Comment by craftninja** [comment]
All tests pass, Nice Job!!
